docs/src/dev-docs/design-docker-compose.md

```
@@ -0,0 +1,101 @@
+   * `provision_queue()`
+   * `provision_redis()`
+   * `provision_results_cache()`
+   * `provision_compression_scheduler()`
+   * `provision_query_scheduler()`
+   * `provision_compression_worker()`
+   * `provision_query_worker()`
+   * `provision_reducer()`
+   * `provision_webui()`
+   * `provision_garbage_collector()`
+
+2. **Environment Generation**: The controller generates a `.env` file with all necessary environment 
```

markdownlint-cli2 (0.17.2)

Trailing spaces
Expected: 0 or 2; Actual: 1

(MD009, no-trailing-spaces)